playpen.py

```diff
@@ -1,8 +1,63 @@
 #!/usr/bin/env python3
 
 import subprocess
+import shlex #for quote() - python3 required
+import re #for re.match()
 
-def execute(version, command, arguments, data=None):
+def execute(version, command, arguments, data=None, env_vars=None):
+    #Note: arguments and env_vars are expected to be of tuple type, not list,
```

Comment:
Oh wow! This actually looks like it's a bit noisy, could the implementation be simplified? We don't really need to do any validation of the command line or anything like that as we're just spawning processes ourselves